### ISSUE TITLE
Improve error recovery for: alias a struct anonymous record

### ISSUE THREAD

**Issue by cartermp** [dotnet/fsharp#6132]
```fsharp
type R = struct {| S: string |}
```
Expected: this works

Actual:
![image](https://user-images.githubusercontent.com/6309070/51406894-47870080-1b0f-11e9-8928-91b3b4c06ede.png)

**Comment by majocha** [comment]
Does `struct` keyword work with records at all?

**Comment by cartermp** [comment]
Nope, for that you use the attribute on the definition. But `struct { |...|}` is the anon record syntax, so it's sort of an odd point.

**Comment by majocha** [comment]
Indeed. Somehow the `struct {| … |}` syntax works in Visual Studio but not in command line fsi on my PC. Hmm.

**Comment by dsyme** [comment]
This is by design.  Parentheses are required.  There is a type declaration form `type R - struct ... end` and that takes precedence.  I'll note this for clarification in the language spec or RFC

    type R = (struct {| S: string |})

**Comment by cartermp** [comment]
Re-opening since the error recovery here is really, really awful:

<img width="330" alt="image" src="https://user-images.githubusercontent.com/6309070/91325762-6f712800-e778-11ea-9a1b-aecdd062aa62.png">

There is no hint at all that you need to add parentheses and it destroys your editor afterwards.

**Comment by dsyme** [comment]
Closing this out as relatively rare and entirely cosmetic
